src/store/index.js

```diff
@@ -162,87 +149,45 @@ export default new Vuex.Store({
     }
   },
   actions: {
-    login ({ commit, dispatch }, user) {
+    login: async function ({ commit, dispatch }, user) {
       commit('setUser', user)
-      dispatch('loadApplicationSettings')
-      dispatch('updateAgencies')
 
-      const doFetch = attr => {
-        fetch(`/api/${attr}`, { credentials: 'include' })
-          .then(r => r.json())
-          .then(data => {
-            const mutation = _.camelCase(`set_${attr}`)
-            commit(mutation, data[attr])
-            if (attr === 'reporting_periods') { // yuck
-              commit('setAllReportingPeriods', data.all_reporting_periods)
-            }
-          })
-      }
-      doFetch('configuration')
-      doFetch('reporting_periods')
+      // to ensure consistent application state, lets block rendering until these complete
+      await dispatch('updateApplicationSettings')
+      await dispatch('updateReportingPeriods')
+      await dispatch('updateAgencies')
+      await dispatch('updateUsersRoles')
```

Comment:
Ooh, I think we can parallelize these and still block render.

```suggestion
      await Promise.all([
        dispatch('updateApplicationSettings'),
        dispatch('updateReportingPeriods'),
        dispatch('updateAgencies'),
        dispatch('updateUsersRoles')
      ])
```